usaspending_api/download/v2/year_limited_downloads.py

```diff
@@ -46,8 +46,35 @@ def process_filters(self, request_data):
             if required_param not in filters:
                 raise InvalidParameterException(f"Missing one or more required body parameters: {required_param}")
 
-        if "agency" not in filters and "agencies" not in filters:
-            raise InvalidParameterException(f"Request must include either 'agency' or 'agencies'")
+        # Replacing agency with agencies
+        if "agency" in filters:
+            if "agencies" not in filters:
+                filters["agencies"] = []
+            if str(filters["agency"]).lower() == "all":
+                toptier_name = "all"
+            else:
+                toptier_name = ToptierAgency.objects.filter(toptier_agency_id=filters["agency"]).values("name").first()
+                if toptier_name is None:
+                    raise InvalidParameterException(f"Toptier ID not found: {filters['agency']}")
+                toptier_name = toptier_name["name"]
+            if "sub_agency" in filters:
+                filters["agencies"].append(
+                    {
+                        "type": "awarding",
+                        "tier": "subtier",
+                        "name": filters["sub_agency"],
+                        "toptier_name": toptier_name,
+                    }
+                )
+                del filters["sub_agency"]
+            else:
+                filters["agencies"].append({"type": "awarding", "tier": "toptier", "name": toptier_name})
+            del filters["agency"]
+
+        if "agencies" in filters:
```

Comment:
Probably cleaner to have else-if here
```suggestion
        elif "agencies" in filters:
```

Reply:
Nevermind, I understand the logic

Reply:
That changes the intent of the PR. The `agency` filter is converted to `agencies`. Then we process `agencies`. If there is no `agencies` filter then it is created, otherwise, it is appended to.